app/components/GenericOauthPlatform.tsx

```diff
@@ -153,15 +148,14 @@ export const GenericOauthPlatform = ({
   }
 
   async function initiateFetchCredential() {
-    if (accessTokenRequest) {
+    if (platform.getProviderProof) {
```

Comment:
@nutrina Heres an attempt at fixing the comments mentioned here: https://github.com/gitcoinco/passport/pull/683#pullrequestreview-1168179057

This should allow moving the proof logic to platforms for platforms that do not use an Oauth window. 

I tried to extract Oauth window logic to the platforms, but was unable to find an elegant way of doing so. imo it would be cleanest to create a separate `GenericOauthPlatform` component that specifically handles creating the Oauth windows, but very open to suggestions 🚀